public/apps/account/account-app.tsx

```diff
@@ -71,7 +72,15 @@ export async function setupTopNavButton(coreStart: CoreStart, config: ClientConf
 
     let shouldShowTenantPopup = true;
 
-    if (tenantSpecifiedInUrl() || getShouldShowTenantPopup() === false) {
+    const dashboardsInfo = await getDashboardsInfoSafe(coreStart.http);
+    const multitenancyEnabled = dashboardsInfo?.multitenancy_enabled && config.multitenancy.enabled;
+    const defaultTenantSet = dashboardsInfo?.default_tenant !== '';
+    if (
+      tenantSpecifiedInUrl() ||
+      getShouldShowTenantPopup() === false ||
+      !multitenancyEnabled ||
+      defaultTenantSet
+    ) {
       shouldShowTenantPopup = false;
     } else {
       // Switch to previous tenant based on localStorage, it may fail due to
```

Comment:
hmm. my suggestion is more of a code clean-up. Rather than assigning it to true and then conditionally assigning it to false, why not directly evaluate and assign the boolean value at declaration.

Reply:
The names are a bit confusing - but `shouldShowTenantPopup` is actually the temporary value that we are setting the value of based on some logic (there is more logic in the else block). Thus I think it would be even more confusing to assign that variable as the conditional check and then change the value in one of the conditional blocks relying on that variable itself?

Reply:
Edit: After in person conversation, I am leaving this as is. The logic is too complicated to do it the suggested way and I want to leave this change small to keep it digestible. Since there is more checks being done and it is still being possibly set as `false` in the `else` block, going to leave this as is until the larger refactor #1761.